spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/pulsar/PulsarProperties.java

```diff
@@ -47,7 +48,7 @@
  * @since 3.2.0
  */
 @ConfigurationProperties("spring.pulsar")
-public class PulsarProperties {
+public class PulsarProperties implements InitializingBean {
```

Comment:
Can we do this differently, please? There are no other `@ConfigurationProperties` classes in Boot's codebase that implement `InitializingBean`. A common place for it is in the consuming auto-configuration.

Reply:
Absolutely. I got pulled away before I could add a self-review comment asking if this is the recommended approach for cross-field validation. I found it in the following (but was skeptical still):

- OAuth2ClientProperties
- DataSourceProperties
- OAuth2AuthorizationServerProperties

I will adjust.

Reply:
Sorry, I failed to find those when searching earlier. On balance I think I still prefer it being done in the auto-configuration, although I feel less strongly about it now.

Reply:
No worries. I'm about to push a commit, I think this is much cleaner. It did have an eyebrow raising effect in its current incantation. :)

Reply:
Done